compiler/rustc_mir/src/borrow_check/type_check/mod.rs

```diff
@@ -749,7 +749,11 @@ impl<'a, 'b, 'tcx> TypeVerifier<'a, 'b, 'tcx> {
                     (&adt_def.variants[VariantIdx::new(0)], substs)
                 }
                 ty::Closure(_, substs) => {
-                    return match substs.as_closure().upvar_tys().nth(field.index()) {
+                    return match substs
+                        .as_closure()
+                        .tupled_upvars_ty()
+                        .tuple_get_field(field.index())
```

Comment:
This seems fine, but wouldn't the old code also work?

Reply:
Ah, I thought they were in the same file. This is faster because the function I added allows O(1) access